packages/server/lib/middleware/access.middleware.ts

```diff
@@ -169,6 +184,44 @@ export class AccessMiddleware {
         }
     }
 
+    private async connectSessionCheck(req: Request): Promise<
+        Result<{
+            account: DBTeam;
+            environment: DBEnvironment;
+            connectSession: ConnectSession;
+        }>
+    > {
+        const authorizationHeader = req.get('authorization');
+
+        if (!authorizationHeader) {
+            return Err('missing_auth_header');
+        }
+
+        const token = authorizationHeader.split('Bearer ').pop();
+
+        if (!token) {
+            return Err('malformed_auth_header');
+        }
+
+        const getConnectSession = await connectSessionService.getConnectSessionByToken(db.knex, token);
```

Comment:
since it's done at every call I would at least check it has the right prefix before looking in DB so you can early exit.

In the case of prefix but wrong key you should also exit but not continue to secretKey check. Because right now it's returning `The provided secret key is not a UUID v4` on invalid sessionToken